### ISSUE TITLE
issue with qlora fine-tuning on Flex GPU

### ISSUE THREAD

**Issue by tsantra** [intel-analytics/BigDL#9308]
Hi,

I am trying to use the Qlora code as provided in the repo on a Sapphire Rapids, Flex GPU machine. 

I was able to run the [qlora_finetuning.py](https://github.com/intel-analytics/BigDL/blob/main/python/llm/example/GPU/QLoRA-FineTuning/qlora_finetuning.py) without any error. 

But the [export_merged_model.py](https://github.com/intel-analytics/BigDL/blob/main/python/llm/example/GPU/QLoRA-FineTuning/export_merged_model.py) is giving me this error:

<img width="887" alt="image" src="https://github.com/intel-analytics/BigDL/assets/45886215/f6935b6a-2fd4-4d83-967d-6933fd40611f">

The command I used to merge the model: 
python ./export_merged_model.py --repo-id-or-model-path < path to llama-2-7b-chat-hf> --adapter_path ./outputs/checkpoint-200 --output_path ./outputs/checkpoint-200-merged

OS : Ubuntu 22
This is my training info:
<img width="872" alt="image" src="https://github.com/intel-analytics/BigDL/assets/45886215/9fe70d04-99be-4497-91df-900764c7a40d">

**Comment by rnwang04** [comment]
Hi, @tsantra Would you mind trying it again after `pip install accelerate==0.23.0` ?

**Comment by tsantra** [comment]
@rnwang04 Thank you.  It worked after installing accelerate=0.23.0 

Is QLora fine-tuning supported on CPU?

**Comment by rnwang04** [comment]
Hi @tsantra ,
1. Now we only support QLora fine-tuning on GPU
2. After you got the merged model (for example `checkpoint-200-merged`), you can use it as a normal huggingface transformer model to do inference on Flex GPU, like https://github.com/intel-analytics/BigDL/tree/main/python/llm/example/GPU/HF-Transformers-AutoModels/Model/llama2

**Comment by rnwang04** [comment]
Hi @tsantra , QLoRA CPU example is updated here(https://github.com/intel-analytics/BigDL/tree/main/python/llm/example/CPU/QLoRA-FineTuning)

**Comment by tsantra** [comment]
Hi @rnwang04 , thank you for your reply! 

Are you using any metric to check for model accuracy after QLora finetuning. I had used my custom dataset for finetuning and my inference results are not good. Model is hallucinating a lot.  Do you have any BKM for fine-tuning? 

Are you also using any profiler to check for GPU memory usage? Do you have any suggestion?

**Comment by tsantra** [comment]
Had closed by mistake.

**Comment by tsantra** [comment]
@rnwang04 GPU finetuning suddenly stopped working and gave Seg Fault. 

<img width="890" alt="image" src="https://github.com/intel-analytics/BigDL/assets/45886215/24711001-95d1-4ffc-a3c7-de618f50041f">

**Comment by rnwang04** [comment]
> @rnwang04 GPU finetuning suddenly stopped working and gave Seg Fault.

Hi @tsantra , have you ever run GPU finetuning successfully ? or you always meet this error?
If you ever run GPU finetuning successfully before, do you make any changes to your script or env settings?

**Comment by rnwang04** [comment]
> Are you using any metric to check for model accuracy after QLora finetuning. 

Not yet.

> I had used my custom dataset for finetuning and my inference results are not good. Model is hallucinating a lot. Do you have any BKM for fine-tuning?

Have you checked your loss curve of finetuning? Is the loss decreasing normally during the finetune process and ultimately stabilizing at a fixed value? What are the approximate train loss and eval loss in the end?

> Are you also using any profiler to check for GPU memory usage? Do you have any suggestion?

I just use "GPU Memory Used" column in "sudo xpu-smi stats -d 0" to check GPU memory usage.

**Comment by shane-huang** [comment]
> @rnwang04 GPU finetuning suddenly stopped working and gave Seg Fault.
> 
> <img alt="image" width="890" src="https://private-user-images.githubusercontent.com/45886215/280564834-24711001-95d1-4ffc-a3c7-de618f50041f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU5Mjc1NjEsIm5iZiI6MTcwNTkyNzI2MSwicGF0aCI6Ii80NTg4NjIxNS8yODA1NjQ4MzQtMjQ3MTEwMDEtOTVkMS00ZmZjLWEzYzctZGU2MThmNTAwNDFmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTIyVDEyNDEwMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFjMmRhY2FhYTJkNTJiZTkwNTE2ZDg4ODgyYzdkYzQyNTQ2NjNlZjAyM2FlZDhkMWNjNDQ2MGJiMzk4MDU3MWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.pakk3vIm1kPP8b6Q83PD_47oqMdHFQk6G7o63u72dnU">

are you running it inside vscode?
